hspec-discover/test/RunSpec.hs

```diff
@@ -84,6 +84,14 @@ spec = do
     it "returns Nothing for invalid spec name" $ do
       fileToSpec "" "foo" `shouldBe` Nothing
 
+    context "when spec does not have a valid module name" $ do
+        it "returns Nothing" $ do
+            fileToSpec "" "flycheck_Spec.hs" `shouldBe` Nothing
```

Comment:
I think both 4 spaces and 2 spaces are valid choices for indentation, but for consistency with the rest of the code, let's stick to 2 spaces here ;)